vowpalwabbit/baseline.cc

```diff
@@ -94,10 +145,17 @@ base_learner* baseline_setup(vw& all)
   data.all = &all;
   if (!all.vm.count("loss_function") || all.vm["loss_function"].as<string>() != "logistic" )
     data.lr_scaling = true;
+  if (all.vm.count("lr_multiplier"))
+    data.lr_multiplier = all.vm["lr_multiplier"].as<float>();
+  data.check_enabled = all.vm.count("check_enabled") > 0;
   data.global_only = all.vm.count("global_only") > 0;
   if (data.global_only)
   { // use a separate global
-    VW::add_constant_feature(all, data.ec);
+    data.ec->indices.push_back(constant_namespace);
+    // shifted index to avoid conflicts
+    data.ec->feature_space[constant_namespace].push_back(1, constant - 17);
```

Comment:
I'm worried about the correctness of this line.   You can't generally subtract an arbitrary constant and 'hit' a parameter.  Instead, you need to obey the stride shift.   See for example this line: https://github.com/JohnLangford/vowpal_wabbit/blob/master/vowpalwabbit/stagewise_poly.cc#L197 which creates an index for a feature in a polynomial learning reduction.

Reply:
iirc (from when I was working on adf in cbify), I think the stride shift and mask aren't set up correctly yet when the setup function here is called. But perhaps I can just have this constant be in a different namespace (maybe a new line in `example.h`) to avoid the arbitrary index altogether here?